docs/workspaces/index.md

```diff
@@ -0,0 +1,136 @@
+        "home": "/home/coder",
+        "args": "--auth none",
+        "gradedFiles": [
+            "starter_code.h",
+            "starter_code.c"
+        ],
+        "syncIgnore": [
+            ".local/share/code-server/"
+        ]
+    }
+}
+```
+
+## Running locally (on Docker)
+
+* First, create an empty directory to use to share job data between containers.
+
+    * This can live anywhere, but needs to be created first and referenced in the docker launch command.
+    * This command is copy-pastable for Windows PowerShell, MacOS, and Linux.
+
+```sh
+mkdir "$HOME/pl_ag_jobs"
+```
+
+* Then, use one of the following `docker run` commands based on your platform.
+
+In MacOS, `cd` to your course directory and copy-paste the following command:
+
+```sh
+docker run -it --rm -p 3000:3000 \
+  -v "$PWD":/course \
+  -v "$HOME/pl_ag_jobs:/jobs" \
+  -e HOST_JOBS_DIR="$HOME/pl_ag_jobs" \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  prairielearn/prairielearn \
+  /PrairieLearn/docker/start_workspace.sh
+```
+
+In Linux, `cd` to your course directory and copy-paste the following command (same as the MacOS command but add the `--add-host` option):
+
+```sh
+docker run -it --rm -p 3000:3000 \
+  -v "$PWD":/course \
+  -v "$HOME/pl_ag_jobs:/jobs" \
+  -e HOST_JOBS_DIR="$HOME/pl_ag_jobs" \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  --add-host=host.docker.internal:172.17.0.1 \ # this line is new vs MacOS
+  prairielearn/prairielearn \
+  /PrairieLearn/docker/start_workspace.sh
+```
+
+In Windows 10 (PowerShell), `cd` to your course directory and copy the following command **except with your own username** in `HOST_JOBS_DIR`:
+
+```powershell
+docker run -it --rm -p 3000:3000 `
+  -v $HOME\pl_ag_jobs:/jobs `
+  -e HOST_JOBS_DIR=/c/Users/Tim/pl_ag_jobs `
+  -v /var/run/docker.sock:/var/run/docker.sock `
+  prairielearn/prairielearn `
+  /PrairieLearn/docker/start_workspace.sh
+```
+
+**Note the following about `HOST_JOBS_DIR` in PowerShell:**
+
+    * Use Unix-style paths (i.e., use `/c/Users/Tim/pl_ag_jobs`, not `C:\Users\Tim\pl_ag_jobs`).
+    * Use the full path rather than $HOME (i.e., use `/c/Users/Tim/pl_ag_jobs`, not `$HOME/pl_ag_jobs`).
+
+### Windows errors and quirks
+
+#### `exec user process caused "no such file or directory"`
+
+This error occurs during grading as a result of an OS new-line incompatibility with the `entrypoint` script in the externally
+graded question:
+
+```sh
+standard_init_linux.go:207: exec user process caused "no such file or directory"
+```
+
+One solution for this is to make a `.gitattributes` files in your PL repository with the line
+`*.sh text eol=lf`. This tells the GitHub client to write the script files in native Linux
+format instead of converting them for Windows (which breaks mapping them back into docker).
+This mimics the [`.gitattributes` file in the main PrairieLearn repo](https://github.com/PrairieLearn/PrairieLearn/blob/master/.gitattributes).
+
+#### `invalid mode: /grade`
```

Comment:
Is this section needed for workspaces? It looks like it will only be a problem for externally graded questions, not workspaces.

Reply:
Replaced the "Windows quirks" section with a note about ensuring `C:` is shared.